core/src/main/resources/lib/hudson/progressive-text.js

```diff
@@ -28,6 +28,12 @@ Behaviour.specify(
         parameters: { start: e.fetchedBytes },
         requestHeaders: headers,
         onComplete: function (rsp) {
+          if (rsp.status >= 500) {
+            setTimeout(function () {
+              fetchNext(e, href, onFinishEvent);
+            }, 1000);
+            return;
```

Comment:
This should additionally add an indicator to the log that Jenkins currently isn't reachable. Not all downtimes are simple restarts for which you can expect the pipeline to resume without trouble.

Reply:
> an indicator to the log

Are you referring to the browser (`console.log`) or appending text to the actual displayed build log? The latter could get messed up if the log contains HTML spanning lines. A different spinner would probably be a better indicator.

Reply:
> Are you referring to the browser (`console.log`) or appending text to the actual displayed build log?

The latter.

> The latter could get messed up if the log contains HTML spanning lines. A different spinner would probably be a better indicator.

Good point, or some other UI that shows up while the server isn't reachable and disappears again when it is. Let's call this one definitely out of scope and I'll try to come up with something.